app/services/cloud_front_request.rb

```diff
@@ -0,0 +1,61 @@
+class CloudFrontRequest
+  def initialize(request)
+    @request = request
+  end
+
+  def client_ip
+    addr = ip_and_port
+    return nil unless addr
+    addr[:ip]
+  end
```

Comment:
This isn't used yet. It's not clear to me whether this will match `request.remote_ip` or not.